business/workloads.go

```diff
@@ -2140,12 +2143,36 @@ func (in *WorkloadService) GetWaypoints(ctx context.Context) models.Workloads {
 
 	waypoints := models.Workloads{}
 	for cluster := range in.userClients {
-		gateways, err := in.GetAllWorkloads(ctx, cluster, config.GatewayLabel)
+		kubeCache, err := in.cache.GetKubeCache(cluster)
 		if err != nil {
-			log.Debugf("GetWaypoints: Error fetching k8s gateway workloads for cluster=[%s]: %s", cluster, err.Error())
-			continue
+			log.Errorf("GetWaypoints: cluster [%s] is not found or is not accessible for Kiali", cluster)
+			return waypoints
 		}
 
+		sel, err := labels.Parse(config.GatewayLabel)
+		if err != nil {
+			log.Errorf("GetWaypoints: bad Gateway selector: %s", err)
+			return waypoints
+		}
+
+		depList := &apps_v1.DeploymentList{}
+		// no namespace selector, default to all namespaces for the cluster
+		selectorOpt := ctrlclient.MatchingLabelsSelector{Selector: sel}
+		if err := kubeCache.List(ctx, depList, selectorOpt); err != nil {
+			log.Errorf("GetWaypoints: error fetching Deployments [%s]: %s", cluster, err)
+			return waypoints
+		}
+		gateways := sliceutil.Map(depList.Items, func(d apps_v1.Deployment) *models.Workload {
+			w := &models.Workload{
+				Pods:     models.Pods{},
+				Services: []models.ServiceOverview{},
+			}
+			w.Cluster = cluster
+			w.Namespace = d.Namespace
+			w.ParseDeployment(&d)
+			return w
+		})
```

Comment:
Can waypoints be other workload types e.g. `Daemonset` or other custom type?

Reply:
I don't think ATM there is an example of a waypoint deployed as something different as a deployment, but, as it can be "customised" (No need to use the actual implementation), I think it would be important to mention it in the docs